### ISSUE TITLE
Finding_DepthFile

### ISSUE THREAD

**Issue by chldydgh4687** [sejongbhaj/SituationClassifier#12]
# Previously...

## 6/4 목표 

- 코드 결과의 depth 맵을 찾을 것
- depth 맵을 우리쪽 데이터셋에 맞게 적용할 것

## 이후 과정

- RGB 디텍션한 GT값을 DEPTH 맵에도 동일시한다.
- 기존 Pretrained 모델을 기반으로 여러분들의 DB에 적용하여 빠르게 disparity image or depth image 정검이 필요함
- GT값의 무게중심을 구해 DEPTH맵의 위치를 파악한다. ( 추후 위험도 분류에 사용될 것임)

**Comment by chldydgh4687** [comment]
### 코드 결과의 depth 파일 찾기
- pxi2pix_model.py 에서 다음함수가 save가 되는지 확인중
~~~
def run_and_save_DAVIS(self, input_, targets, save_path):
        assert (self.num_input == 3)
        input_imgs = autograd.Variable(input_.cuda(), requires_grad=False)

        stack_inputs = input_imgs

        prediction_d, pred_confidence = self.netG.forward(stack_inputs)
        pred_log_d = prediction_d.squeeze(1)
        pred_d = torch.exp(pred_log_d)

        if not os.path.exists(save_path):
            os.makedirs(save_path)

        for i in range(0, len(targets['img_1_path'])):

            youtube_dir = save_path + targets['img_1_path'][i].split('/')[-2]

            if not os.path.exists(youtube_dir):
                os.makedirs(youtube_dir)

            saved_img = np.transpose(
                input_imgs[i, :, :, :].cpu().numpy(), (1, 2, 0))

            pred_d_ref = pred_d.data[i, :, :].cpu().numpy()

            output_path = youtube_dir + '/' + \
                targets['img_1_path'][i].split('/')[-1]
            print(output_path)
            disparity = 1. / pred_d_ref
            disparity = disparity / np.max(disparity)
            disparity = np.tile(np.expand_dims(disparity, axis=-1), (1, 1, 3))
            saved_imgs = np.concatenate((saved_img, disparity), axis=1)
            saved_imgs = (saved_imgs*255).astype(np.uint8)

            imsave(output_path, saved_imgs)
~~~

**Comment by chldydgh4687** [comment]
saved_imgs = (save_imgs).astype(np.unt8) 으로 수정
- 검은 화면이 저장이됨. 영상으로치면 255 대입함으로써 원래색으로 복구되는듯함. 
---
np.concatenate 부분을 주석처리( saved와 disparity 병합한 것이 아닌 saved_imgs만 뽑아봄)
- saved_imgs 제대로 regression network가 들어가지 않는 것을 발견한듯함.
---
다음과 같이 코드를 변경하여 disparity 만 뽑았다. (@nhk9680 맞는지 확인 부탁 )
아! 교수님 전에 disparity 누르셨단 얘기 이해가 됬습니다. 감사합니다. (@unizard)
- 병합한 depth가 맞는듯함 코드를 변형시켜 우리쪽 데이터셋에 맞게 적용하겠음.
~~~
            disparity = 1. / pred_d_ref
            disparity = disparity / np.max(disparity)
            disparity = np.tile(np.expand_dims(disparity, axis=-1), (1, 1, 3))
         #   saved_imgs = np.concatenate((saved_img, disparity), axis=1)
            saved_imgs = (disparity)
~~~

![wetwetwew](https://user-images.githubusercontent.com/11037567/58844432-2552e100-86b2-11e9-98f3-01edd809684a.PNG)

**Comment by chldydgh4687** [comment]
## depth 맵을 우리쪽 데이터셋에 맞게 적용할 코드 수정

**Comment by unizard** [comment]
먼저 입력데이터만 바꿔서 넣어봐

**Comment by chldydgh4687** [comment]
![436436436364643](https://user-images.githubusercontent.com/11037567/58860804-e2abfb80-86e7-11e9-804c-650c00ea198d.PNG)
![23523523](https://user-images.githubusercontent.com/11037567/58862700-39b3cf80-86ec-11e9-9a0e-fb3bf4d153ef.PNG)


동일한 픽쳐 한장인데..해상도도 DAVIS 해상도로 똑같이 변환하여 시도했는데 너무 차이가 없는 것 같습니다.
다른 픽쳐로 시도해보고 두번째 논문으로 시도해보겠습니다.

**Comment by chldydgh4687** [comment]
![123456](https://user-images.githubusercontent.com/11037567/58862416-982c7e00-86eb-11e9-93d2-74c4425463d0.PNG)
![123467](https://user-images.githubusercontent.com/11037567/58862420-995dab00-86eb-11e9-973d-cbb8718d4900.PNG)

5m 반경 가까운 물체는 어느정도 잡히는 것 같습니다. (@unizard)

**Comment by unizard** [comment]
@chldydgh4687 RGB+Depth overlay 시켜봐야 할것 같은데...

**Comment by chldydgh4687** [comment]
한번 알아보겠습니다.

**Comment by unizard** [comment]
@chldydgh4687  https://kr.mathworks.com/help/images/ref/imfuse.html

**Comment by unizard** [comment]
> ![436436436364643](https://user-images.githubusercontent.com/11037567/58860804-e2abfb80-86e7-11e9-804c-650c00ea198d.PNG)
> ![23523523](https://user-images.githubusercontent.com/11037567/58862700-39b3cf80-86ec-11e9-9a0e-fb3bf4d153ef.PNG)
> 
> 동일한 픽쳐 한장인데..해상도도 DAVIS 해상도로 똑같이 변환하여 시도했는데 너무 차이가 없는 것 같습니다.
> 다른 픽쳐로 시도해보고 두번째 논문으로 시도해보겠습니다.

이정도면 굳이 두번째 논문 안해도 됨.. 눈에 안보이는게 정상임
너희 데이터 셋이 아니라서 그러니까 괜춘

**Comment by chldydgh4687** [comment]
넵! 으..감사합니다 ㅠ 제가 MATLAB은 깔아야되서 이게 시간이 걸리는 것으로 알고있는데 깔면서 코드는 짜서 있는 친구에게 돌려달라 요청하겠습니다.

**Comment by unizard** [comment]
@chldydgh4687 깔지말고, 그냥 몇장만 교수님한테 보내 아니면 깃에 올려줘.. 갈길이 멀었음

**Comment by unizard** [comment]
@chldydgh4687  일단 위에 오버레이 결과 나오면, RGB BBbox GT가지고 장애물까지의 대략적인 평균 거리 값(relative depth) 을 계산해서 영상 박스 위에 그려서 보자. 그리고 남훈이 쪽 완성되면 둘이 합치면 됨

그리고 현재 너의 결과 영상이 depth 인지 disparity 인지 확인해보고

**Comment by chldydgh4687** [comment]
저 메일 보냈는데 결과 영상 DEPTH인지 DISPARITY 인지 확인해보겠습니다.

**Comment by unizard** [comment]
![image](https://user-images.githubusercontent.com/11758940/58874377-a6869400-8703-11e9-8279-d8d901289623.png)

```
clc; close all; clear all;

im1 = imread('s5_Depth.jpg');
im2 = imread('s5_RGB.jpg');
[hh,ww,ch] = size(im2);
im1_ = imresize(im1, [hh,ww]);
im3 = imfuse(im1_,im2);

figure; imshow(im1);
figure; imshow(im2);
figure; imshow([im1_, im2, im3]); 
```

**Comment by unizard** [comment]
> 저 메일 보냈는데 결과 영상 DEPTH인지 DISPARITY 인지 확인해보겠습니다.

영상데이터 뿌려보니, disparity 임

**Comment by chldydgh4687** [comment]
다시 찾아야할까요...ㅠ

**Comment by nhk9680** [comment]
????????????????????

**Comment by unizard** [comment]
@nhk9680 @chldydgh4687 남훈이에게 설명했으니, 용호는 남훈이에게 둘어~ 아무래도 용호가 disparity와 depth 의 상관관계를 모르는듯 하네~

https://docs.opencv.org/3.4/dd/d53/tutorial_py_depthmap.html

**Comment by chldydgh4687** [comment]
신경쓰이게 해드려서 죄송합니다 더 정진하겠습니다
